cmd/util/sign.go

```diff
@@ -12,92 +12,93 @@ package main
 import (
 	"crypto"
 	"crypto/ecdsa"
+	"crypto/ed25519"
 	"crypto/elliptic"
 	_ "crypto/sha256"
 	_ "crypto/sha512"
-	"encoding/asn1"
 	"fmt"
 	"io"
-	"math/big"
 )
 
-type ECDSASigner struct {
+// Signer represents an structure holding the signing information
+type Signer struct {
 	bugs  *CertificateBugs
 	curve elliptic.Curve
 	hash  crypto.Hash
-	priv  *ecdsa.PrivateKey
+	priv  crypto.PrivateKey
 	rand  io.Reader
+	ecdsa bool
 }
 
-func maybeCorruptECDSAValue(n *big.Int, typeOfCorruption BadValue, limit *big.Int) *big.Int {
```

Comment:
Why was this removed?

Reply:
I'm not really sure why you had it in the first place. In order to support different algorithms, you can use the crypto.Signer interface where this check does not make sense anymore, as the signature returned is already a byte array. Why was it needed if the signature is generated by the golang code?

Reply:
So, my question is more: why was the check there before?

Reply:
Ah no, this wasn't a check. This was to malform the ECDSA signature, which we planned to use in future fuzzing tests. Perhaps it should be named better ...

Reply:
Ahhh hahaha.. now it makes sense ;) Ok, I can create a func that makes the byte array empty or changes some bytes, if corruption is needed.

Reply:
Yeah, we have the `Bugs CertificateBugs` struct to control this, and we hope to adding much more to that.

Reply:
I added now some corruption, but it can be extended.